diff --git a/build/lib/populate-sw-template.js b/build/lib/populate-sw-template.js
index c142f5945f7d06c16a1c009568e15e1c10233888..718545807874518cc205a2db022c78cfc6254c65 100644
--- a/build/lib/populate-sw-template.js
+++ b/build/lib/populate-sw-template.js
@@ -56,7 +56,7 @@ function populateSWTemplate({ cacheId, cleanupOutdatedCaches, clientsClaim, dire
             clientsClaim,
             disableDevLogs,
             importScripts,
-            manifestEntries,
+            manifestEntries: manifestEntries.filter(({ url }) => url !== '404.html').map(({ url, revision }) => ({ url: encodeURI(url), revision })),
             navigateFallback,
             navigateFallbackDenylist,
             navigateFallbackAllowlist,
